statsd/statsd.go

```diff
@@ -43,6 +46,26 @@ func SetBackend(b Backend) {
 	currentBackend = b
 }
 
+// ErrUnknownBackend is returned when the statsd backend implementation is not known.
+var ErrUnknownBackend = fmt.Errorf("unknown statsd backend type")
+
+// SetBackendFor returns the appropriate Backend for the given implementation and host.
+func SetBackendFor(impl, addr, prefix string, tags ...string) error {
```

Comment:
How about decoupling the "Set" and add a `NewBackend(impl, addr, prefix string, tags ...string) (Backend, error)`?